manubot/cite/csl_item.py

```diff
@@ -1,6 +1,201 @@
+import re
+import logging
+
 from manubot.cite.citekey import standardize_citekey, infer_citekey_prefix, is_valid_citekey
+from manubot.cite.citeproc import remove_jsonschema_errors, get_jsonschema_csl_validator
+
+
+csl_item_type_fixer = {
+    'journal-article': 'article-journal',
+    'book-chapter': 'chapter',
+    'posted-content': 'manuscript',
+    'proceedings-article': 'paper-conference',
+    'standard': 'entry',
+    'reference-entry': 'entry',
+}
+
+class CSL_Item(dict):
```

Comment:
I wonder if this should extend `collections.OrderedDict`? Plain dicts have been ordered since 3.6, but some software still doesn't treat them as such (but does treat OrderedDicts as ordered). I think extending `dict` is fine unless testing determines that some non-determinism or sorting by key gets introduced.

Reply:
I did not think of OrderedDict, but I experimented with UserDict. Unfortunately UserDict had problems qith serialisation, I could not get it to work with json.dump(). 

I used dict because it is used now for csl_item.

Reply:
extending dict is correct. I looked into this more and basically `collections.OrderedDict` should be used as little as possible given that in the future it should never be needed. Therefore if we need `collections.OrderedDict` for something to work, we can cast to it when needed but not before.